lib/stale.js

```diff
@@ -80,7 +80,68 @@ module.exports = class Stale {
 
     const query = queryParts.join(' ')
     const days = this.getConfigValue(type, 'days') || this.getConfigValue(type, 'daysUntilStale')
-    return this.search(type, days, query)
+    const results = await this.search(type, days, query)
+
+    const pinned = await this.getPinnedNumbers(type)
+    const nonPinned = results.data.filter(issue => pinned.includes(issue.number))
+    return {
+      ...results,
+      data: {
+        ...results.data,
+        total_count: nonPinned.length,
+        items: nonPinned,
+      }
+    }
+  }
+
+  async getPinnedNumbers(type) {
+    if (type === 'pulls') {
+      // Pull Requests cannot be pinned
+      return []
+    } else if (type !== 'issues') {
+      throw new Error(`Unknown type: ${type}. Valid types are 'pulls' and 'issues'`)
+    }
+
+    const { owner, repo } = this.config
+
+    try {
+      // GitHub's v3 REST API doesn't support Pinned Issues; v4 GraphQL API does
+      const {data, errors} = await this.github.graphql(
+        `
+          query Issues($owner: String!, $repo: String!) {
+            repository(owner: $owner, name: $repo) {
+              pinnedIssues(first: 100) {
+                nodes {
+                  number
+                }
+                pageInfo {
+                  # This should always be false, as only 3 issues can be pinned at the same time
+                  hasNextPage
+                }
+              }
+            }
+          }
+        `,
```

Comment:
I originally extracted this into a constant, but decided it might be clearer to have it nearby.